src/keybinding/fish.rs

```diff
@@ -0,0 +1,88 @@
+use color_eyre::eyre::{Result, eyre};
+use crossterm::event::{KeyCode, KeyEvent, KeyModifiers};
+
+/// Represents a single Fish keybinding entry
+pub struct FishKeyFields {
+    pub modifier: String,
+    pub keycode: String,
+}
+
+pub fn fish_keyevent_to_fields(ev: KeyEvent) -> Result<FishKeyFields> {
+    use KeyCode::*;
+    use KeyModifiers as KM;
+
+    // --- Determine modifier string ---
+    let ctrl = ev.modifiers.contains(KM::CONTROL);
+    let alt = ev.modifiers.contains(KM::ALT);
+    let shift = ev.modifiers.contains(KM::SHIFT);
+
+    let modifier = match (ctrl, alt, shift) {
+        (false, false, false) => "",
+        (true, false, false) => "ctrl-",
+        (false, true, false) => "alt-",
+        (false, false, true) => "",
+        (true, true, false) => "ctrl-alt-",
+        (true, false, true) => "ctrl-",
+        (false, true, true) => "alt-",
+        (true, true, true) => "ctrl-alt",
+    }
+    .to_string();
+
+    // --- Determine keycode string ---
+    let keycode = match ev.code {
+        Backspace => "backspace".into(),
+        Enter => "enter".into(),
+        Esc => "escape".into(),
+        Tab => "tab".into(),
+        Char(' ') => "space".into(),
+        Char(c) => c.into(),
+        Delete => "delete".into(),
+        Insert => "insert".into(),
+        Home => "home".into(),
+        End => "end".into(),
+        PageUp => "pageup".into(),
+        PageDown => "pagedown".into(),
+        Up => "up".into(),
+        Down => "down".into(),
+        Left => "left".into(),
+        Right => "right".into(),
+        F(n) => format!("F{}", n),
```

Comment:
According to the docs, these should be lowercase, not sure if that makes a difference.
```suggestion
        F(n) => format!("f{}", n),
```